### PR TITLE
fix(terraform_plan): Fix tf plan nested modules

### PR DESCRIPTION
**Pull request by achiar99**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

Fix resources created by modules for tf_plan

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/terraform/plan_runner.py [hunk 1]
```diff
@@ -247,14 +248,18 @@ def run_block(self, entities,
...
             raw_context['definition_path'] = entity[CustomAttributes.BLOCK_NAME].split('.')
         return raw_context, None
 
-    def get_entity_context(self, definition_path, full_file_path):
-        entity_id = ".".join(definition_path)
+    def get_entity_context(self, definition_path, full_file_path, entity):
+        resource_type = definition_path[0]
+        resource_name = ''
+        if resource_type in entity.keys():
+            resource_name = definition_path[1]
```
Comment (YaaraVerner) [inline checkov/terraform/plan_runner.py:261]: why do we need this?

File: checkov/terraform/plan_utils.py [hunk 2]
```diff
@@ -65,21 +65,28 @@ def create_definitions(
...
         entities = definition.get(block_type, [])
         for entity in entities:
             context_parser = parser_registry.context_parsers[block_type]
             definition_path = context_parser.get_entity_context_path(entity)
-            entity_id = ".".join(definition_path)
+
+            resource_type = definition_path[0]
+            resource_name = ''
+            if resource_type in entity.keys():
+                resource_name = definition_path[1]
```
Comment (YaaraVerner) [inline checkov/terraform/plan_utils.py:79]: also here

**Review by YaaraVerner — APPROVED** [review]

File: checkov/terraform/plan_runner.py [hunk 3]
```diff
@@ -247,14 +248,18 @@ def run_block(self, entities,
...
+        entity_id = entity.get(resource_type, {}).get(resource_name).get(TF_PLAN_RESOURCE_ADDRESS)
```
Comment (gruebel) [inline checkov/terraform/plan_runner.py:262]:
```suggestion
        entity_id = entity.get(resource_type, {}).get(resource_name, {}).get(TF_PLAN_RESOURCE_ADDRESS)
```

File: checkov/terraform/plan_utils.py [hunk 4]
```diff
@@ -65,21 +65,28 @@ def create_definitions(
...
+
+            entity_id = entity[resource_type][resource_name].get(TF_PLAN_RESOURCE_ADDRESS)
```
Comment (gruebel) [inline checkov/terraform/plan_utils.py:81]:
```suggestion
            entity_id = entity.get(resource_type, {}).get(resource_name, {}).get(TF_PLAN_RESOURCE_ADDRESS)
```

File: checkov/terraform/plan_runner.py (hunk 1 above)
Comment (gruebel) [reply checkov/terraform/plan_runner.py:261]: probably we could simplify it, when the `defintion_path` has a length of at least 2
```python
        resource_type = definition_path[0]
        resource_name = definition_path[1]
```

File: checkov/terraform/plan_utils.py (hunk 2 above)
Comment (gruebel) [reply checkov/terraform/plan_utils.py:79]: same here

**Review by gruebel — APPROVED** [review]
🍪

**Pull request merged**